### PR TITLE
DOC: Create initial documentation

### PR DESCRIPTION
**Pull request by daikitag**
Create initial documentation for tstrait.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
This is a bit of a mess I'm afraid @daikitag, I think we're trying to do too much at once. I'll make some notes on the diff in a minute

File: docs/Makefile [hunk 1]
```diff
@@ -14,3 +15,25 @@ dist:
 
 clean:
 	rm -fR _build
+=======
```
Comment (jeromekelleher) [inline docs/Makefile:18]: There's a merge conflict here

File: docs/source/_templates/autosummary/class.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{% extends "!autosummary/class.rst" %}
```
Comment (jeromekelleher) [inline docs/source/_templates/autosummary/class.rst:1]: I don't think this file should be commited? Also there's a bunch of svg files that probably don't need to go in (like the example tree sequence?)

File: docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import pathlib
+import sys
+
```
Comment (jeromekelleher) [inline docs/source/conf.py:3]: Why conf.py? I think we should follow the stucture used by msrpime etc using jupyterbook

File: requirements/CI-docs/requirements.txt [hunk 4]
```diff
@@ -1,10 +1,10 @@
-jupyter-book==0.15.1
```
Comment (jeromekelleher) [inline requirements/CI-docs/requirements.txt:1]: These are deliberately pinned to known working versions

**Review by jeromekelleher — COMMENTED** [review]
I don't think the new file organisation helps here @daikitag, it's just creating a lot of files and directories and makes it harder to find what you're looking for. I think it would be simpler to follow the pattern used in msprime, where possible.

**Comment by jeromekelleher** [comment]
Is there some way we can split this up into smaller changes so that we get the docs build working, firstly? Currently it's all quite broken looking and it's not obvious where to start fixing it.

**Comment by jeromekelleher** [comment]
Can you describe what you're trying to do here, and explain the different dependencies etc you are bringing in?

**Comment by daikitag** [comment]
Thank you for your feedback. [docs/source/_templates/autosummary/class.rst](https://github.com/tskit-dev/tstrait/pull/74/files/0a3427c2faaadce21cdef106328bddb522bb2346#diff-8829a387fe850486a1e3531b7a97bf8f873535bc57a1302d2c211e290e32024e) is used to structure the API documentation (Sphinx automatically documents all methods inside a class, such as __init__), and the svg files are used in the title page.
I created different folders to put the markdown files in corresponding folders (user, reference, etc), and I used setup.py, as it is the method used in Sphix.
But I will restructure the files, such that the HTML documentation can be built by using jupyter-book.

**Pull request closed without merging**